src/prefect/agent.py

```diff
@@ -256,31 +257,40 @@ async def check_for_cancelled_flow_runs(self):
 
         self.logger.debug("Checking for cancelled flow runs...")
 
-        work_queue_names = set()
-        async for work_queue in self.get_work_queues():
-            work_queue_names.add(work_queue.name)
+        work_queue_filter = (
+            WorkQueueFilter(name=WorkQueueFilterName(any_=list(self.work_queues)))
+            if self.work_queues
+            else None
+        )
 
+        work_pool_filter = (
+            WorkPoolFilter(name=WorkPoolFilterName(any_=[self.work_pool_name]))
+            if self.work_pool_name
+            else None
+        )
```

Comment:
If a work pool isn't provided, we should only look at queues in the default-agent-work pool. Otherwise, the agent could cancel flow runs in other work pools. 

There's a constant for this work pool name, but it's in the serverside code, so we might not want to import it here. We could have a client-side version of the constant too.
```suggestion
        work_pool_filter = (
            WorkPoolFilter(name=WorkPoolFilterName(any_=[self.work_pool_name]))
            if self.work_pool_name
            else WorkPoolFilter(name=WorkPoolFilterName(any_=["default_agent_pool"]))
        )
```